### DISCUSSION TITLE
Transition FAST v7 to OpenFAST

### DISCUSSION THREAD

**Discussion by omaudhuit** [OpenFAST/openfast#905] [Q&A]
Hello everyone,

Firstly, thank you all in advance for any feedback that is provided on this subject. 

We manufacture a 50kW turbine where the original simulations were done using a compiled version of FAST v7. 

To simulate fault events (grid loss, short circuit) - defined in the IEC 61400 standards - which were modeled in the Simulink model, the following was done:

 - The tip brake parameters (TTpBrDp1/2) used in the .fst file are parsed by FAST.
 - Within the bladeddllinterface.f90 (lines 505/506), (avrSWAP()) = TTpBrDp()) values are assigned.
 - This information is then passed to the dll (in order to be used as Event ID and Time in the controller model). 

We are now in the process of transitioning over to OpenFAST and would like to potentially find a less complicated way of simulating such fault events. Are there dedicated 'channels' that execute the above in the source code? or is there a way for us to create our own 'channels' for the same purpose? 

The attachment shows the above flow (.fst file -> .f90 -> c code to compile the dll).

Please let us know if anything needs any clarification or if more visuals are needed. Looking forward to the feedback.

[Discussion_2.pdf](https://github.com/OpenFAST/openfast/files/7458439/Discussion_2.pdf)

**Comment by jjonkman** [comment]
Dear @omaudhuit,

Actually, the old tip brake functionality of FAST v7 was not ported to FAST v8 or OpenFAST.  There are still some remnants of the old tip-brake calculations in the source code, but the overall functionality would have to implemented to enable tip-brake calculations in OpenFAST.

Best regards,
